src/stitcher/image_loader.rs

```diff
@@ -92,11 +101,16 @@ pub fn find_images(directory_path: impl AsRef<Path>) -> Result<Vec<PathBuf>, Ima
 ///  - The directory is invalid or does not contain any images.
 ///  - The directory does not contain any jpg, jpeg, png, or webp images.
 ///  - An image cannot be opened.
-pub fn load_images(directory_path: &str, width: Option<u32>) -> Result<RgbImage, ImageLoaderError> {
-    let dimensions = find_images(directory_path)?
-        .into_iter()
-        .map(|image| image_dimensions(image).map_err(|e| ImageLoaderError::from(e)))
+pub fn load_images(paths: &[impl AsRef<Path>], width: Option<u32>) -> Result<RgbImage, ImageLoaderError> {
```

Comment:
In my opinion, `load_images` should just load a given set of images, and `find_images` should be used instead to find the images to load.